src/modules/roc_sndio/target_sox/roc_sndio/player.cpp

```diff
@@ -71,13 +65,32 @@ bool Player::open(const char* name, const char* type) {
         return false;
     }
 
+    roc_log(LogInfo,
+            "player:"
+            " bits=%lu out_rate=%lu in_rate=%lu ch=%lu",
+            (unsigned long)output_->encoding.bits_per_sample,
+            (unsigned long)output_->signal.rate, (unsigned long)out_signal_.rate,
+            (unsigned long)output_->signal.channels);
+
     return true;
 }
 
 void Player::stop() {
     stop_ = 1;
 }
 
+void Player::start(pipeline::IReceiver* input) {
+    input_ = input;
+    core::Thread::start();
+}
+
+size_t Player::get_sample_rate() const {
+    if (output_) {
+        return size_t(output_->signal.rate);
+    }
+    return 0;
```

Comment:
If output_ is null, it means that open() was not called or failed, and the user called get_sample_rate() anyway. It looks like a bug in the user code, so I think it's better to panic here instead of silently returning zero.

Reply:
Fixed

Reply:
Fixed

Reply:
Fixed

Reply:
Fixed